src/LearningObjectSubmission/SubmissionRouteDriver.ts

```diff
@@ -24,41 +26,74 @@ export function initialize({
 }) {
   async function submit(req: Request, res: Response) {
     try {
-      const id = req.params.learningObjectId;
-      const username = req.user.username;
+      const learningObjectId = req.params.learningObjectId;
+      const userId = req.params.userId;
+      const user: UserToken = req.user;
       const collection = req.body.collection;
 
       await submitForReview({
         dataStore,
         fileManager,
-        username,
-        id,
+        user,
+        learningObjectId,
+        userId,
         collection,
       });
 
       res.sendStatus(200);
     } catch (e) {
-      console.error(e);
-      res.status(500).send(e);
+      const { code, message } = mapErrorToResponseData(e);
+      res.status(code).json({message});
+    }
+  }
+
+  async function fetchSubmission(req: Request, res: Response) {
+    try {
+      const learningObjectId = req.params.learningObjectId;
+      const userId = req.params.userId;
+      const collection = req.query.collection;
+      const hasSubmission = req.query.hasSubmission;
+      const emailVerified = req.user.emailVerified;
+
+      if (!collection || !hasSubmission) {
+        res.status(501).json({message: 'Not Implemented'});
```

Comment:
Something like this might make debugging easier on the client-side:

```suggestion
        res.status(501).json({message: 'This route currently requires both the collection and hasSubmission query parameters.'});
```